### ISSUE TITLE
drive: permission invites get re-sent on occasion

### ISSUE THREAD

**Issue by bobheadxi** [ubclaunchpad/rocket2#497]
I can't seem to fetch permissionDetails to determine if a permission is inherited - something about the difference between personal drives and team drives? https://developers.google.com/drive/api/v3/reference/permissions#resource

One solution might be to not delete permissions at all => https://github.com/ubclaunchpad/rocket2/issues/497#issuecomment-705993259

**Comment by bobheadxi** [comment]
#504 has a hack for this: https://github.com/ubclaunchpad/rocket2/commit/35e1209682d9906e7ccf7bf0ee64c601de0084ee

**Comment by bobheadxi** [comment]
35e1209 did not fix this, because we do not recreate shares that cascade.

**Comment by bobheadxi** [comment]
Reverting https://github.com/ubclaunchpad/rocket2/commit/35e1209682d9906e7ccf7bf0ee64c601de0084ee actually caused me to start seeing the issue described in #510 - I periodically receive invites for the projects folder now as well. I do not receive invites for folders like Exec and the main Launch Pad folder, however

**Comment by bobheadxi** [comment]
#540 disabled share deletion, so I've updated the issue to indicate cascading permissions deletion might not be the issue here.

Thought: it could be that we need to page through results, given that this only happens for the Projects folder which is our largets.

**Comment by bobheadxi** [comment]
Some more findings: we do not have shared drives, that is limited to a pricing tier above what we have, so the permissions inheritance approach (#499) is a no-go

**Comment by bobheadxi** [comment]
As of  https://github.com/ubclaunchpad/rocket2/commit/bc9a6d846c413ece61d0fc1217fe8e067a975ebb:

```
�2020-10-17T01:41:51.026112486Z {Time: 2020-10-17 01:41:51, Level: [INFO], function: gcp.ensure_drive_permissions():144, message: Found 42 permissions for all that do not require updating}
�2020-10-17T01:43:01.543056021Z {Time: 2020-10-17 01:43:01, Level: [INFO], function: gcp.ensure_drive_permissions():168, message: Created 50 permissions for all}
```

There still seems to be a lot of creations that are not getting seen

**Comment by bobheadxi** [comment]
I think we're finally clean through a series of direct-to-master pushes oops https://github.com/ubclaunchpad/rocket2/compare/55b48e57485e3f52e79e2c13ec93052d67ecd46f...ffc67b3adbea4c55b28cb4914bc0cdb91da3085a

tl;dr Rocket was failing to identify who has already received shares because:
* emails are case-insensitive - but google magically turns emails into upper-cased things (presumably based on what they have on their end), so anyone who cased their emails in Rocket differently from how Google things they have it cased would have their permissions constantly deleted and created
* dots don't matter in emails (!!!!!) so anyone who uses dots in their emails might have their emails magically changed by google to ones without dots (or vice versa)

(╯°□°）╯︵ ┻━┻

If you accept the most recent drive invite, you should be good to go now! Sorry about the spam :sweat_smile:
